Unable to locate package libenchant1c2a

I've always been able to install the list of extensions.
Today, (when upgrading my Docker image from PHP 7.4.21 to 7.4.22) I do get an error on the `enchant` package.
I'm using the `php:7.4.22-fpm` Docker image for PHP.

See the logs below:
```
#7 0.267 install-php-extensions v.1.2.57                                                                                                                                                    
#7 1.872 Updating channel "pecl.php.net"                                                                                                                                                    
#7 2.108 Channel "pecl.php.net" is up to date                                                                                                                                               
#7 2.232 Get:1 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
#7 2.271 Get:2 http://deb.debian.org/debian bullseye InRelease [113 kB]
#7 2.301 Get:3 http://deb.debian.org/debian bullseye-updates InRelease [36.8 kB]
#7 2.386 Get:4 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [25.8 kB]
#7 2.466 Get:5 http://deb.debian.org/debian bullseye/main amd64 Packages [8178 kB]
#7 4.191 Fetched 8397 kB in 2s (4082 kB/s)
#7 4.191 Reading package lists...
#7 5.610 FAILED TO LIST THE WHOLE PACKAGE LIST FOR
#7 5.610 libenchant1c2a libfreetype6 libjpeg62-turbo ^libpng[0-9]+-[0-9]+$ libxpm4 ^libwebp[0-9]+$ libc-client2007e ^libicu[0-9]+$ libmcrypt4 libpq5 libpq5 ^libtidy-?[0-9][0-9.\-]*(deb[0-9])?$ ^libxslt1\.1$ ^libzip[0-9]$ ^libmbedtls[0-9]*$ 
#7 5.610 
#7 5.610 COMMAND OUTPUT:
#7 5.610 Reading package lists...
#7 5.610 Building dependency tree...
#7 5.610 Reading state information...
#7 5.610 E: Unable to locate package libenchant1c2a
#7 5.610 
#7 5.610 E: apt-get failed
```

When I remove the `enchant` from the list, all extensions get installed without errors. However, I would like to have the `enchant` extension installed as well.

Not sure if I have the same/similar issue or not, but my (previously working) usage of `docker-php-extension-installer` has started failing in a similar manner when attempting to (re-)build a docker image today:

## Input
```
RUN chmod +x /usr/local/bin/install-php-extensions && sync && \
    install-php-extensions \
        bcmath \
        mysqli \
        pdo_mysql \
        intl \
        ldap \
        gd \
        soap \
        tidy \
        xsl \
        zip \
        exif \
        gmp
```

## Output
```
install-php-extensions v.1.2.57
Err:1 http://security.debian.org/debian-security buster/updates InRelease
  Temporary failure resolving 'security.debian.org'
Err:2 http://deb.debian.org/debian buster InRelease
  Temporary failure resolving 'deb.debian.org'
Err:3 http://deb.debian.org/debian buster-updates InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists...
W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://security.debian.org/debian-security/dists/buster/updates/InRelease  Temporary failure resolving 'security.debian.org'
W: Failed to fetch http://deb.debian.org/debian/dists/buster-updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
FAILED TO LIST THE WHOLE PACKAGE LIST FOR
^libicu[0-9]+$ libfreetype6 libjpeg62-turbo ^libpng[0-9]+-[0-9]+$ libxpm4 ^libwebp[0-9]+$ ^libtidy-?[0-9][0-9.\-]*(deb[0-9])?$ ^libxslt1\.1$ ^libzip[0-9]$ ^libmbedtls[0-9]*$ 

COMMAND OUTPUT:
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libfreetype6
E: Unable to locate package libjpeg62-turbo
E: Unable to locate package ^libpng[0-9]+-[0-9]+$
E: Couldn't find any package by glob '^libpng[0-9]+-[0-9]+$'
E: Couldn't find any package by regex '^libpng[0-9]+-[0-9]+$'
E: Unable to locate package libxpm4
E: Unable to locate package ^libwebp[0-9]+$
E: Couldn't find any package by glob '^libwebp[0-9]+$'
E: Couldn't find any package by regex '^libwebp[0-9]+$'
E: Unable to locate package ^libtidy-?[0-9][0-9.\-]*(deb[0-9])?$
E: Couldn't find any package by glob '^libtidy-?[0-9][0-9.\-]*(deb[0-9])?$'
E: Couldn't find any package by regex '^libtidy-?[0-9][0-9.\-]*(deb[0-9])?$'
E: Unable to locate package ^libxslt1\.1$
E: Couldn't find any package by glob '^libxslt1\.1$'
E: Couldn't find any package by regex '^libxslt1\.1$'
E: Unable to locate package ^libzip[0-9]$
E: Couldn't find any package by glob '^libzip[0-9]$'
E: Couldn't find any package by regex '^libzip[0-9]$'
E: Unable to locate package ^libmbedtls[0-9]*$
E: Couldn't find any package by glob '^libmbedtls[0-9]*$'
E: Couldn't find any package by regex '^libmbedtls[0-9]*$'

E: apt-get failed
The command '/bin/sh -c chmod +x /usr/local/bin/install-php-extensions && sync &&     install-php-extensions         bcmath         mysqli         pdo_mysql         intl         ldap         gd         soap         tidy         xsl         zip         exif         gmp' returned a non-zero code: 1
##[error]The command '/bin/sh -c chmod +x /usr/local/bin/install-php-extensions && sync &&     install-php-extensions         bcmath         mysqli         pdo_mysql         intl         ldap         gd         soap         tidy         xsl         zip         exif         gmp' returned a non-zero code: 1
##[error]The process '/usr/bin/docker' failed with exit code 1
```

Does this look like the same or a related issue, or should I open a new issue for this?

--- a/data/special-requirements
+++ b/data/special-requirements
@@ -1,4 +1,3 @@
-enchant !7.2-alpine3.12 !7.3-alpine3.12 !7.3-alpine3.13 !7.4-alpine3.12 !7.4-alpine3.13
 parallel zts
 pdo_sqlsrv !7.0-alpine3.7 !7.1-alpine3.7 !7.1-alpine3.8 !7.2-alpine3.7 !7.2-alpine3.8 !7.3-alpine3.8
 pthreads zts

--- a/install-php-extensions
+++ b/install-php-extensions
@@ -53,8 +53,10 @@ resetIFS() {
 }
 
 # Set these variables:
-# - DISTRO containing distribution name (eg 'alpine', 'debian')
-# - DISTO_VERSION containing distribution name and its version(eg 'alpine@3.10', 'debian@9')
+# - DISTRO containing the distribution name (eg 'alpine', 'debian')
+# - DISTRO_VERSION_NUMBER containing the distribution version (eg '3.14' for Alpine, 11 for Debian)
+# - DISTRO_VERSION containing the distribution name and its version(eg 'alpine@3.14', 'debian@11')
+# - DISTRO_MAJMIN_VERSION always containing a number representing the distribution version (eg 314 for Alpine, 1100 for Debian)
 setDistro() {
 	if ! test -r /etc/os-release; then
 		printf 'The file /etc/os-release is not readable\n' >&2
@@ -492,16 +494,33 @@ buildRequiredPackageLists() {
 				;;
 			enchant@alpine)
 				if test $DISTRO_MAJMIN_VERSION -ge 312; then
-					buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent enchant2"
-					buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile enchant2-dev"
+					if test $PHP_MAJMIN_VERSION -ge 800; then
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent enchant2"
+						buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile enchant2-dev"
+					else
+						# The system provides libenchant2, supported since PHP 8.0: we need to build libenchant1 on our own
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent glib aspell-libs libhunspell"
+						buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile glib-dev aspell-dev hunspell-dev"
+					fi
 				else
 					buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent enchant"
 					buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile enchant-dev"
 				fi
 				;;
 			enchant@debian)
-				buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libenchant1c2a"
-				buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile libenchant-dev"
+				if test $DISTRO_VERSION_NUMBER -ge 11; then
+					if test $PHP_MAJMIN_VERSION -ge 800; then
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libenchant-2-2"
+						buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile libenchant-2-dev"
+					else
+						# The system provides libenchant2, supported since PHP 8.0: we need to build libenchant1 on our own
+						buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent aspell-en libhunspell-1.7-0"
+						buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile libglib2.0-dev libaspell-dev libhunspell-dev"
+					fi
+				else
+					buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libenchant1c2a"
+					buildRequiredPackageLists_volatile="$buildRequiredPackageLists_volatile libenchant-dev"
+				fi
 				;;
 			event@alpine)
 				buildRequiredPackageLists_persistent="$buildRequiredPackageLists_persistent libevent $buildRequiredPackageLists_libssl"
@@ -1460,6 +1479,32 @@ installBundledModule() {
 EOF
 			fi
 			docker-php-ext-configure dba --with-db4
+			;;
+		enchant)
+			installBundledModule_tmp=0
+			if test $PHP_MAJMIN_VERSION -lt 800; then
+				case "$DISTRO" in
+					alpine)
+						if test $DISTRO_MAJMIN_VERSION -ge 312; then
+							installBundledModule_tmp=1
+						fi
+						;;
+					debian)
+						if test $DISTRO_VERSION_NUMBER -ge 11; then
+							installBundledModule_tmp=1
+						fi
+						;;
+				esac
+			fi
+			if test $installBundledModule_tmp -eq 1 && ! test -f /usr/lib/libenchant.so && ! test -f /usr/local/lib/libenchant.so; then
+				# We need to install libenchant1 from source
+				installBundledModule_src="$(getPackageSource https://github.com/AbiWord/enchant/releases/download/enchant-1-6-1/enchant-1.6.1.tar.gz)"
+				cd -- "$installBundledModule_src"
+				./configure
+				make -j$(getProcessorCount)
+				make install
+				cd - >/dev/null
+			fi
 			;;
 		gd)
 			if test $PHP_MAJMIN_VERSION -le 506; then

